test/test_optim.py

```diff
@@ -311,10 +311,9 @@ def _test_complex_optimizer(self, optimizer_constructor):
         complex_opt = optimizer_constructor(complex_param)
         real_opt = optimizer_constructor(real_param)
 
-        for i in range(3):
+        for _ in range(3):
             complex_param.grad = torch.randn_like(complex_param)
-            real_param.grad = torch.view_as_real(complex_param.grad)
-
+            real_param.grad = torch.view_as_real(complex_param.grad.clone())
```

Comment:
grad gets mutated internally by the optimizer.

Reply:
What? Which optimizer does that??

Reply:
rprop multi tensor path:
https://github.com/pytorch/pytorch/blob/eff28d61c9c961e6c2724b78f57441ee2e3e40cb/torch/optim/rprop.py#L225-L226

Reply:
Not intended probably 🤔

Reply:
Ho wow I missed that :o
Yes this should be out of place!
Can you fix it here please?